sopel/modules/url.py

```diff
@@ -36,6 +40,44 @@ class UrlSection(StaticSection):
     exclusion_char = ValidatedAttribute('exclusion_char', default='!')
 
 
+class TitleParser(HTMLParser):
+    def __init__(self):
+        try:
+            HTMLParser.__init__(self, convert_charrefs=False)
```

Comment:
In the commit I made 2 months ago to make the code behave identically on Python 2 and 3, I had to set that to `False` because the setting doesn't exist in older `HTMLParser`. The test suite I used was a long list of links we collected on IRC that were broken in the past and I used a virtualenv for each version of Python to get them all working as best I could or at least looking the same in each. There are some that are hopeless cases, not even rendered correctly by browsers, but the other 99% work great now in both Python 2 and 3.

Reply:
In the commit I made 2 months ago to make the code behave identically on Python 2 and 3, I had to set that to `False` because the setting doesn't exist in older `HTMLParser`. The test suite I used was a long list of links we collected on IRC that were broken in the past and I used a virtualenv for each version of Python to get them all working as best I could or at least looking the same in each. There are some that are hopeless cases, not even rendered correctly by browsers, but the other 99% work great now in both Python 2 and 3.

Reply:
I can accept that some cases will be hopeless—some pages are just flat-out broken, there's no way around it. But http://ircbot.science/titles/htmlentityunicode/ is as well-formed as you can get, and titles like that should work. `convert_charrefs` may be unavailable on py2, but what about using `decode` from `sopel.web` to convert entities? Is that doable?